### DISCUSSION TITLE
How to Fetch API data in background

### DISCUSSION THREAD

**Discussion by Ranger11Danger** [widgetti/solara#297] [Q&A]
i was wondering what solutions everyone has come up with for querying an api on a set timer to update data on the page? I have been using the `use_thread` method but it cant run more than 1 thread at a time, so i would have to query all my different endpoints in the same thread which I feel is slower. Just wanted to see if anyone has come up with any other solutions

**Comment by maartenbreddels** [comment]
Hi,

sorry for the late reply!

> but it cant run more than 1 thread at a time
Why is that the case?
You could use use_thread in combination with a asyncio loop to have the thread run multiple async tasks that fetch your data maybe?


Regards,

Maarten
